src/libraries/System.Private.CoreLib/src/System/IO/StreamReader.cs

```diff
@@ -245,27 +245,67 @@ protected override void Dispose(bool disposing)
             {
                 return;
             }
-            _disposed = true;
+            try
+            {
+                if (!_disposed && disposing)
```

Comment:
You already checked _disposed above.
```suggestion
                if (disposing)
```